src/Formats/ProtobufSchemas.h

```diff
@@ -28,14 +28,36 @@ class FormatSchemaInfo;
 class ProtobufSchemas : private boost::noncopyable
 {
 public:
-    static ProtobufSchemas & instance();
+    enum class WithEnvelope
+    {
+        // Return descriptor for a top-level message with a user-provided name.
+        // Example: In protobuf schema
+        //   message MessageType {
+        //     string colA = 1;
+        //     int32 colB = 2;
+        //   }
+        // message_name = "MessageType" returns a descriptor. Used by IO
+        // formats Protobuf and ProtobufSingle.
+        No,
```

Comment:
I am wondering if it is okay in CH to use scoped enums for boolean parameters? They are a bit more readable in method invocations on the callers end.

Example:

`func(true, false) // or: func(/*with_envelope = */ true, /*with_length = */ false);`

vs.

`func(WithEnvelope::Yes, WithLength::No);`

Reply:
It's ok to use it